src/sparsematrix.jl

```diff
@@ -44,6 +44,58 @@ function SparseMatrixCSC(m::Integer, n::Integer, colptr::Vector, rowval::Vector,
     SparseMatrixCSC{Tv,Ti}(m, n, colptr, rowval, nzval)
 end
 
+SparseMatrixCSC(m, n, colptr::ReadOnly, rowval::ReadOnly, nzval::Vector) =
+    SparseMatrixCSC(m, n, copy(parent(colptr)), copy(parent(rowval)), nzval)
+
+struct FixedSparseCSC{Tv,Ti<:Integer} <: AbstractSparseMatrixCSC{Tv,Ti}
+    m::Int                  # Number of rows
+    n::Int                  # Number of columns
+    colptr::ReadOnly{Ti,Vector{Ti}} # Column i is in colptr[i]:(colptr[i+1]-1)
+    rowval::ReadOnly{Ti,Vector{Ti}} # Row indices of stored values
+    nzval::Vector{Tv}       # Stored values, typically nonzeros
+
+    function FixedSparseCSC{Tv,Ti}(m::Integer, n::Integer,
+                            colptr::ReadOnly{Ti,Vector{Ti}},
+                            rowval::ReadOnly{Ti,Vector{Ti}},
+                            nzval::Vector{Tv}) where {Tv,Ti<:Integer}
+        sparse_check_Ti(m, n, Ti)
+        _goodbuffers(Int(m), Int(n), parent(colptr), parent(rowval), nzval) ||
+            throw(ArgumentError("Invalid buffers for FixedSparseCSC construction n=$n, colptr=$(summary(colptr)), rowval=$(summary(rowval)), nzval=$(summary(nzval))"))
+        new(Int(m), Int(n), colptr, rowval, nzval)
+    end
+end
+@inline _is_fixed(::FixedSparseCSC) = true
+FixedSparseCSC(m::Integer, n::Integer,
+    colptr::ReadOnly{Ti, Vector{Ti}},
+    rowval::ReadOnly{Ti, Vector{Ti}},
+    nzval::Vector{Tv}) where {Tv,Ti<:Integer} =
+    FixedSparseCSC{Tv,Ti}(m, n, colptr, rowval, nzval)
+FixedSparseCSC{Tv,Ti}(m::Integer, n::Integer, colptr::Vector{Ti}, rowval::Vector{Ti}, nzval::Vector{Tv}) where {Tv,Ti} =
+    FixedSparseCSC{Tv,Ti}(m, n, ReadOnly(colptr), ReadOnly(rowval), nzval)
+FixedSparseCSC(m::Integer, n::Integer, colptr::Vector{Ti}, rowval::Vector{Ti}, nzval::Vector{Tv}) where {Tv,Ti} =
```

Comment:
is this method dangerous? it's pretty handy and since SparseMatrixCSC edits the vectors, i guess this is safe.